facebook/delphiFacebook/R/contingency_tables.R

```diff
@@ -0,0 +1,1068 @@
+  end_date <- as.Date(date_range[[2]])
+  date_pattern = "^[0-9]{4}-[0-9]{2}-[0-9]{2}.*[.]csv$"
+  youtube_pattern = ".*YouTube[.]csv$"
+  
+  filenames <- list.files(path=params$input_dir)
+  filenames <- filenames[grepl(date_pattern, filenames) & !grepl(youtube_pattern, filenames)]
+  
+  file_end_dates <- as.Date(substr(filenames, 1, 10))
+  file_start_dates <- file_end_dates
+  
+  # Only keep files with data that falls at least somewhat between the desired
+  # start and end range dates.
+  filenames <- filenames[
+    !(( file_start_dates < start_date & file_end_dates < start_date ) | 
+        ( file_start_dates > end_date & file_end_dates > end_date ))]
+  
+  return(filenames)
+}
+
+
+#' Run the contingency table production pipeline
+#'
+#' See the README.md file in the source directory for more information about how to run
+#' this function.
+#'
+#' @param params    Params object produced by read_params
+#' @param aggregations Data frame with columns `name`, `var_weight`, `metric`,
+#'   `group_by`, `compute_fn`, `post_fn`. Each row represents one aggregate
+#'   to report. `name` is the aggregate's base column name; `var_weight` is the 
+#'   column to use for its weights; `metric` is the column of `df` containing the
+#'   response value. `group_by` is a list of variables used to perform the 
+#'   aggregations over. `compute_fn` is the function that computes
+#'   the aggregate response given many rows of data. `post_fn` is applied to the
+#'   aggregate data after megacounty aggregation, and can perform any final
+#'   calculations necessary.
+#'
+#' @return none
+#' 
+#' @importFrom parallel detectCores
+#' 
+#' @export
+run_contingency_tables <- function(params, aggregations)
+{
+  params <- update_params(params)
+  
+  cw_list <- produce_crosswalk_list(params$static_dir)
+  archive <- load_archive(params)
+  msg_df("archive data loaded", archive$input_data)
+  
+  input_data <- load_responses_all(params)
+  input_data <- filter_responses(input_data, params)
+  msg_df("response input data", input_data)
+  
+  input_data <- merge_responses(input_data, archive)
+  data_agg <- create_data_for_aggregatation(input_data)
+  
+  data_agg <- filter_data_for_aggregatation(data_agg, params, lead_days = 12)
+  data_agg <- join_weights(data_agg, params, weights = "step1")
+  msg_df("response data to aggregate", data_agg)
+  
+  ## Set default number of cores for mclapply to the total available number,
+  ## because we are greedy and this will typically run on a server.
+  if (params$parallel) {
+    cores <- detectCores()
+    
+    if (is.na(cores)) {
+      warning("Could not detect the number of CPU cores; parallel mode disabled")
+      params$parallel <- FALSE
+    } else {
+      options(mc.cores = cores)
+    }
+  }
+  
+  if ( "cids" %in% params$output )
+  {
+    write_cid(data_agg, "part_a", params)
+  }
+  if ( "archive" %in% params$output )
+  {
+    update_archive(input_data, archive, params)
+  }
```

Comment:
If the contingency table generator only runs periodically, I'd assume that the daily `run.R` will handle all CID and archive generation, and this script doesn't need to do separate work on the archive or CID lists. @krivard can confirm, but I think the CID generation is pretty tightly integrated into her shell pipeline for producing weights and aggregates and won't need to also use this script

Reply:
Correct. The daily pipeline run uses the following output settings in `params.json`:

```
  "output": [
    "cids",
    "individual",
    "covidalert",
    "archive",
    "community"
  ],
```